src/PostgresOutbox/Database/Run.cs

```diff
@@ -16,14 +17,16 @@ public static class Run
         await command.ExecuteNonQueryAsync(ct);
     }
 
-    public static async Task Execute(
-        string connectionString,
-        string sql,
-        CancellationToken ct)
+    public static async Task EnsureTableExists(this NpgsqlDataSource dataSource, string tableName, CancellationToken ct)
```

Comment:
**KUDOS:** Good call for making it accept data source instead of creating it internally.